micrometer-core/src/test/java/io/micrometer/core/instrument/composite/CompositeMeterRegistryTest.java

```diff
@@ -445,4 +448,31 @@ void stopTrackingMetersThatAreRemoved() {
         counting.publish();
         assertThat(counting.count(functionCounter)).isEqualTo(1);
     }
+
+    @RepeatedTest(100)
+    void meterRegistrationShouldWorkConcurrently() throws InterruptedException {
+        this.composite.add(this.simple);
+
+        String meterName = "test.counter";
+        String tagName = "test.tag";
+
+        int count = 10000;
+        int tagCount = 100;
+        ExecutorService executor = Executors.newFixedThreadPool(10);
+        for (int i = 0; i < count; i++) {
+            int tagValue = i % tagCount;
+            executor.execute(() -> {
+                Counter counter = Counter.builder(meterName).tag(tagName, String.valueOf(tagValue))
+                        .register(this.composite);
+                counter.increment();
+            });
+        }
+        executor.shutdown();
+        assertThat(executor.awaitTermination(1L, TimeUnit.DAYS)).isTrue();
```

Comment:
Let's not block for 1 day in a test because this will require manual intervention to stop builds, or rely on the build system to cancel build jobs that are running for too long. I know there should be no reason the tasks don't complete in a timely manner, but I think it'll be better to be safe here with a more reasonable wait time like 1 second.

Reply:
@shakuzen Thanks for the feedback! I changed it in ed62622 as suggested.